### ISSUE TITLE
Output location or access issue? 

### ISSUE THREAD

**Issue by CESTechSupport** [nanoporetech/dorado#304]
Hi everyone, 

It looks like the Dorado basecaller is running properly, but we are not sure where the output file gets located by default. 
For example, if we run the basecalling for the POD5 files on Desktop, we see nothing new generated on the desktop when the run is completed:

~~~~~~~~~~~~~~~~
./dorado.exe basecaller dna_r10.4.1_e8.2_400bps_sup@v4.2.0 C:\Users\Me\Desktop --emit-fastq
~~~~~~~~~~~~~~~~

If we input specific output location, we see this kind of error:

~~~~~~~~~~~~~~~~
./dorado.exe basecaller dna_r10.4.1_e8.2_400bps_sup@v4.2.0 C:\Users\Me\Desktop --emit-fastq -r > C:\Users\Me\Desktop
~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~
out-file : Access to the path 'C:\Users\Me\Desktop' is denied.
At line:1 char:1
+ ./dorado.exe basecaller dna_r10.4.1_e8.2_400bps_sup@v4.2.0 C:\Users\M ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OpenError: (:) [Out-File], UnauthorizedAccessException
    + FullyQualifiedErrorId : FileOpenFailure,Microsoft.PowerShell.Commands.OutFileCommand
~~~~~~~~~~~~~~~~

I am pretty sure that we allowed a full access and ran the powershell as administrator (we are using Windows 11). 
What might be the issue here?

**Comment by sklages** [comment]
`dorado` writes its results on stdout, in your terminal window. In your example, `dorado` expects the POD5 data being located in `C:\Users\Me\Desktop`. Is that intented? 

Is there any output in your terminal window from where you have started `dorado`? If so, what do you see?

In your second example you direct `dorado` to recursively search for POD5 data starting in `C:\Users\Me\Desktop`. 
The error explains that you cannot write (redirect the terminal output) to a directory (name). Instead you must provide a filename.

Just some hints:
- locate your POD5 data, e.g. `C:\Users\Me\Desktop\run\pod5` or something similar. Try to avoid using the Desktop folder directly, instead organise your data in a separate directory.
- add `dorado.exe` to your $PATH or run it with absolute path, e.g. `/path/to/dorado.exe`
- create and go to a results directory, e.g. `D:\data\run_id\basecalls`
- run `dorado` from there, e.g. `/path/to/dorado.exe basecaller dna_r10.4.1_e8.2_400bps_sup@v4.2.0 C:\Users\Me\Desktop\run\pod5 --emit-fastq -r > basecalls.fastq`

**Comment by CESTechSupport** [comment]
Thank you so much for your comment.

Yes, we wanted to use the POD5 data located on our Desktop and that is intended. 
We placed 5 POD5 files and they are all different samples. 

We referenced your hints and ran this code: 

~~~~~~
./dorado.exe basecaller dna_r10.4.1_e8.2_400bps_sup@v4.2.0 C:\Users\Me\Desktop\Test --emit-fastq -r > basecalls.fastq
~~~~~~

It looks like the output file `basecalls.fastq` gets generated in the folder where `dorado.exe` is located.
Is there any way to select the output directory? Also, the basecalled file did not have a quality string, is this typical? 

And we need 5 separate fastq files for each samples, but there was only one file in the directory (I think it basecalled all 5 POD5 files in one fastq). 
Is there any way to basecall them separately?

**Comment by tijyojwad** [comment]
> Is there any way to select the output directory?

As @sklages suggested, you can create a new folder called `basecalls` and redirect the output to `dorado.exe basecaller ... --emit-fastq > basecalls\basecalls.fastq` and that should write the data into a file in the basecalls subfolder.

> Also, the basecalled file did not have a quality string, is this typical?

That's surprising. Can you post the first few lines of the base called output?

> Is there any way to basecall them separately?

By default dorado grabs all POD5s within a directory and calls them (this is because many long running experiments producing multiple pod5 files for the same sample). To organize your data by sample, you can create a separate subfolder for each sample and move the appropriate POD5s into them (e.g. sample_1/file1.pod5, sample_2/file2.pod5, etc). Then you can invoke dorado on each sample folder separately to generate their fastqs.

**Comment by sklages** [comment]
> It looks like the output file `basecalls.fastq` gets generated in the folder where `dorado.exe` is located. 

Yes, because you are running basecalling  from where `dorado` is located and redirecting the output to a file in just that same directory.

>Is there any way to basecall them separately?

Yes, organise your data in a separate directory structure, by run, by sample, whatever is approbiate. Then run basecalling on each of the per-run or per-sample POD5 directories ..

**Comment by CESTechSupport** [comment]
Thank you very much for the suggestions. 
We are now good with running Dorado with your suggestions. 

We use the de-novo workflow `wf-clone-validation` from `EPI2ME` and it is keep showing that the quality string is missing from FASTQ files basecalled using Dorado (but it works fine with guppy basecalled files). 
The FASTQ file looks fine to me - maybe it is workflow related issue? 

`@84b033d5-df87-4eb1-8666-29ffb5e63f76
ATGTTTTTACCTCTACTGGTTGTTGAAGAACCGGGTGTTTAACCGCATAGTTCTGCATGATGGGTTAGGTTTTCGCATTCTCGTGAAACACTTTCGCGTTTTTCGTGCGCCGCTCATTCTTCCAGGACCTTCTTCAACAGGCCAGCCATTACGCTCGTCATCAAAATCACTCGCATCAACCAAACCGTTATTCATTCGTGATTGCGCCTGAGCGAGACGAAATACGCGATCGCTGTTAAAAGGACAATTACAAACAGGAATCGAATGCAACCGGCGCAGGAACACTGCCAGCGCATCAACAATATTTTCACCTGAATCAGGATATTCTTCTAATACCTGGAATACTATTCCGGGGATCGCAGTGGTGAGTAACCATGCATCATCAGGAGTACGGATAAAATGCTTGATGGTCGGAAGAGGCATAAATTCCGTCAGCCAGTTCATCTTGACCATCTCATCTGTAACATCATTGGCAACGCTACCTTTGCCATGTTTCAGAAACAACTCTGGCGCATCGGGCTTCCCATACAAGCGATAGATTGTCGCACCTGATTGCCGACATTATCGCGAGCCCATTTATACCCATATAAATCAGCATCCATGTTGGAATTTAATCGCGGCCTCGACGTTTCCCGTTGAATATGGCTCATAACACCCCTTGTATTACTGTTTATGTAAGCAGACAGTTTTATTGTTCATGATGATATATTTTTATCTTGTGCAATGTAACATCAGAGATTTTGAGACACGGGCCAGAGCTGCATCGCGCGTTTCGGTGATGACGGTGAAAACCTCTGACACATGCAGCTCCCGGAGACGGTCACAGCTTGTCTGTAAGCGGATGCCGGGAGCAAGCAGCCCGTCAGGGCGCGTCAGCGGGTGTTGGCGGGTGTCGGGGCTGGCTTAACTATGCGGCATCAGAGCGGTCTGTACTGAGAGTGCACCATATGCGGTGTGAAATACCGCACAGATGCGTAAGGAGAAAATACCGCATCAGGCGCCATTCGCCATTCAGGCTGCGCAACTGTTGGGAAGGGCGATCGGTGCGGGCCTCTTCGCTACTGCGCAAGTCACCGAAAGGGGGATGTGCTGCAAGGCGATTAAGTTGGGTAACGCGAGGGTTTCCCAGTCACGACGTTGTAAAACGACGGCCAGAAAGCTTGCTCTTCTTT
+
)*))*'%%$%&&&'%%%%%&)$$$$#$$$$%%%&&'????=>GFEKGJKIJNNGJSMIIHNESSJIQJHGJII9.+***)))**02BEI10000GHGGJCGOSMFEFIIHNIFG=<=?440)(((''((((&''''FGGSKISPJSNSSJJSJNSSLSLSSLSSSSHIHPLNSHSPGLHGHIMJSLIJJSSIFSIGMSSHJSMKIPSLJHSKILSIKKISIISMLRSMPEKFISOSKSSLONJISSQHKLSFSLRSJGJLNMPSSHSSSJHG?>:2,&&&+,BAFJSGSSSSLKSHSIMGQLSSLSSSHSLLHLHKSJSGKLHSISKPSKSKJLJKSIMK;:,(((**)()0<DGHSHA@===@ASIRSSHJSMKSSHKGKMJPJNJLLHHHHGLFOIIHLGHGMLHHGSHJQGJFHSSFHSSSSMSJLSHSMSKIP9855($$$$%%'-CDJHQHGOKKNSMHRIKLSLMKILFGJHLSILHGSSSSSMQJKOLMOSGKSJNGHJJISSJJJHSSIHSJIFKNSILSLGFHMHJHFSHIISSILIKSSSNSKSB@E?BHHMJLE-,,,-HDEK?CC@2F?>?IFIJJSGNOLISHSMHIFJGEKORFEHLSSSNGHFDHJHOLEDECEJI7753899=BBDIHGFHGJIFFGIGGIGHPGDDDFFMIHFFKLFGHHFGGEDDDDCEGMKMHGHGGFJIGEGHFLLEJFHSLSQSQGOGLGQGHFEGGGGFFEEGHKGSGRSHEFFHJNFCGFEF?=99:<=ADBOGEKGFHJFCCCCIINSFHNFEFIJNIJHPGIILJMHGSSGKDFSS4//GHGHIIIJHRSFMJHLNLSIKNNGLHQILEIJFGCBFG=...-+*+-(467>CDDDIHQDDIIGMGSSLFMGJFSLSMLH;;;<FJJOJGMGSSLJEEHEKKISDA9988..+*)(&%&&((*--.3312222DCCFFGIHFHEJFSIHFDA33224640.../JKGJSMSSSSSLIGDDB>=<:9:44443599GIAHGSNIREAA@@CEGLIHHKSKKHKIMCDCB??>?@44334:776567>55556>C65544222-****>JNMGJHIKKSSSGOMIIJSFSRSONSGOKIHCBA@++++)+/0(((()888DKDHSSHLJISJFG3332222367@BD@@@@@JSSFD9:+`

**Comment by sklages** [comment]
@CESTechSupport - This fastq record looks badly formatted and should probably look this way:
```
@84b033d5-df87-4eb1-8666-29ffb5e63f76 
ATGTTTTTACCTCTACTGGTTGTTGAAGAACCGGGTGTTTAACCGCATAGTTCTGCATGATGGGTTAGGTTTTCGCATTCTCGTGAAACACTTTCGCGTTTTTCGTGCGCCGCTCATTCTTCCAGGACCTTCTTCAACAGGCCAGCCATTACGCTCGTCATCAAAATCACTCGCATCAACCAAACCGTTATTCATTCGTGATTGCGCCTGAGCGAGACGAAATACGCGATCGCTGTTAAAAGGACAATTACAAACAGGAATCGAATGCAACCGGCGCAGGAACACTGCCAGCGCATCAACAATATTTTCACCTGAATCAGGATATTCTTCTAATACCTGGAATACTATTCCGGGGATCGCAGTGGTGAGTAACCATGCATCATCAGGAGTACGGATAAAATGCTTGATGGTCGGAAGAGGCATAAATTCCGTCAGCCAGTTCATCTTGACCATCTCATCTGTAACATCATTGGCAACGCTACCTTTGCCATGTTTCAGAAACAACTCTGGCGCATCGGGCTTCCCATACAAGCGATAGATTGTCGCACCTGATTGCCGACATTATCGCGAGCCCATTTATACCCATATAAATCAGCATCCATGTTGGAATTTAATCGCGGCCTCGACGTTTCCCGTTGAATATGGCTCATAACACCCCTTGTATTACTGTTTATGTAAGCAGACAGTTTTATTGTTCATGATGATATATTTTTATCTTGTGCAATGTAACATCAGAGATTTTGAGACACGGGCCAGAGCTGCATCGCGCGTTTCGGTGATGACGGTGAAAACCTCTGACACATGCAGCTCCCGGAGACGGTCACAGCTTGTCTGTAAGCGGATGCCGGGAGCAAGCAGCCCGTCAGGGCGCGTCAGCGGGTGTTGGCGGGTGTCGGGGCTGGCTTAACTATGCGGCATCAGAGCGGTCTGTACTGAGAGTGCACCATATGCGGTGTGAAATACCGCACAGATGCGTAAGGAGAAAATACCGCATCAGGCGCCATTCGCCATTCAGGCTGCGCAACTGTTGGGAAGGGCGATCGGTGCGGGCCTCTTCGCTACTGCGCAAGTCACCGAAAGGGGGATGTGCTGCAAGGCGATTAAGTTGGGTAACGCGAGGGTTTCCCAGTCACGACGTTGTAAAACGACGGCCAGAAAGCTTGCTCTTCTTT 
+
)*))*'%%$%&&&'%%%%%&)$$$$#$$$$%%%&&'????=>GFEKGJKIJNNGJSMIIHNESSJIQJHGJII9.+***)))**02BEI10000GHGGJCGOSMFEFIIHNIFG=<=?440)(((''((((&''''FGGSKISPJSNSSJJSJNSSLSLSSLSSSSHIHPLNSHSPGLHGHIMJSLIJJSSIFSIGMSSHJSMKIPSLJHSKILSIKKISIISMLRSMPEKFISOSKSSLONJISSQHKLSFSLRSJGJLNMPSSHSSSJHG?>:2,&&&+,BAFJSGSSSSLKSHSIMGQLSSLSSSHSLLHLHKSJSGKLHSISKPSKSKJLJKSIMK;:,(((**)()0<DGHSHA@===@ASIRSSHJSMKSSHKGKMJPJNJLLHHHHGLFOIIHLGHGMLHHGSHJQGJFHSSFHSSSSMSJLSHSMSKIP9855($$$$%%'-CDJHQHGOKKNSMHRIKLSLMKILFGJHLSILHGSSSSSMQJKOLMOSGKSJNGHJJISSJJJHSSIHSJIFKNSILSLGFHMHJHFSHIISSILIKSSSNSKSB@E?BHHMJLE-,,,-HDEK?CC@2F?>?IFIJJSGNOLISHSMHIFJGEKORFEHLSSSNGHFDHJHOLEDECEJI7753899=BBDIHGFHGJIFFGIGGIGHPGDDDFFMIHFFKLFGHHFGGEDDDDCEGMKMHGHGGFJIGEGHFLLEJFHSLSQSQGOGLGQGHFEGGGGFFEEGHKGSGRSHEFFHJNFCGFEF?=99:<=ADBOGEKGFHJFCCCCIINSFHNFEFIJNIJHPGIILJMHGSSGKDFSS4//GHGHIIIJHRSFMJHLNLSIKNNGLHQILEIJFGCBFG=...-+*+-(467>CDDDIHQDDIIGMGSSLFMGJFSLSMLH;;;<FJJOJGMGSSLJEEHEKKISDA9988..+*)(&%&&((*--.3312222DCCFFGIHFHEJFSIHFDA33224640.../JKGJSMSSSSSLIGDDB>=<:9:44443599GIAHGSNIREAA@@CEGLIHHKSKKHKIMCDCB??>?@44334:776567>55556>C65544222-****>JNMGJHIKKSSSGOMIIJSFSRSONSGOKIHCBA@++++)+/0(((()888DKDHSSHLJISJFG3332222367@BD@@@@@JSSFD9:+
```
Fastq header (@...) at its own line, followed by sequence data on its own line, the quality header (+) again on its own line, followed by the quality string, on its own line :-) - see https://labs.epi2me.io/notebooks/Introduction_to_fastq_file.html

**Comment by tijyojwad** [comment]
Yeah fastq looks okay to me. Could be a wf related issue. Might be better if you post this issue in https://github.com/epi2me-labs/wf-clone-validation/issues ?
